src/bdd_2d.jl

```diff
@@ -0,0 +1,381 @@
+            detIstart = nDet+1
+            detIinc = -1
+        end
+
+        
+        deltaDetm = detm[detIstart+detIinc]- detm[detIstart] # Mapped detector length()
+        deltaPixm = pixm[1,2]- pixm[1,1]   # Mapped pixel length()    
+        
+
+        # For each row
+        for row in 1:nPix
+            
+            rowm = pixm[row,:] # Get first mapped row from image --> changed to column vec
+            reconTmp = zeros(1,nPix)
+
+            detInd = detIstart
+            pixInd = pixIstart            
+            
+            # Find first detector overlap mapped with pixel mapped [Case 1]
+            if (detm[detInd]-rowm[pixIstart]<=deltaDetm)
+                while ((detm[detInd]-rowm[pixIstart]<=deltaDetm))            
+                    detInd = detInd + detIinc            
+                end
+            else
+            # Find first pixel overlap mapped with detector mapped [Case 2]           
+                if (detm[detIstart]-rowm[pixInd]>deltaPixm)
+                    while (detm[detIstart]-rowm[pixInd]>deltaPixm)            
+                        pixInd = pixInd + pixIinc           
+                    end
+                end
+            end
+            
+            detm = vec(detm)
+            Ppj = integrate1D(sinogram[proj,:],deltaDetm)
+
+            if detm[1] > detm[2] #if descending
+                Pdk = Spline1D(reverse(detm), reverse(vec(Ppj)); k=1, bc="zero")(vec(rowm))
+            else 
+                Pdk = Spline1D(detm, vec(Ppj); k=1, bc="zero")(vec(rowm))
+            end
+            
+            while (pixInd < nPix+1)
+                reconTmp[pixInd] = reconTmp[pixInd]+(Pdk[pixInd+c1]-Pdk[pixInd+c2])/deltaDetm
+                pixInd = pixInd + pixIinc 
+            end
+            
+            reconImgTmp[row,:] = reconTmp .* L
+            
+        end # Row loop 
+        
+        if ((angleCase == 1)||(angleCase == 3))
+            reconImg = reconImg + reconImgTmp 
+        else
+            reconImg = reconImg + reverse(reconImgTmp, dims = 2)'
+        end 
+           
+    end # Projection loop
+    
+    reconImg = reconImg / length(theta) 
+    return reconImg
+
+end # endfunc
+
+
+
+## Integrate function
+function integrate1D(p_v::Vector,pixelSize)
+
+    n_pixel = length(p_v)
+    
+    P_x = 0
+    Ppj = zeros(n_pixel+1)
+    
+    for pj in 1:n_pixel
+
+       P_x += p_v[pj] * pixelSize
+       
+       Ppj[pj+1] = P_x
+       
+    end
+
+    return Ppj
+
+end
+
+## Map Y
+# Function that map detector | pixel boundaries onto Y axis()
+function mapp2y(x1,y1,x2,y2)
+    y = y1-x1*(y1-y2)/(x1-x2)
+    return y
+end
+
+## Map X
+# Function that map detector | pixel boundaries onto X axis()
+function mapp2x(x1,y1,x2,y2)
+    x = -(y1)*(x1-x2)/(y1-y2)+x1
+    return x
+end
+
+## Draw 2D
+# Function to draw geometry
+function drawGeo(tubeX,tubeY,detX,detY)
+    plot(tubeX,tubeY,markershape = :asterisk)
+    plot!([detX[1],detX[end]],[detY[1],detY[end]])
+end
```

Comment:
```suggestion
end
=#
```
It was good for testing, but not needed in final version...